### DISCUSSION TITLE
Save user and creation date for folders

### DISCUSSION THREAD

**Discussion by Raffy27** [directus/directus#19032] [Feature Requests]
### Summary

Currently, folders are created as entries in the `directus_folders` table. Files have `uploaded_on` and `uploaded_by` fields, which makes it easy to differentiate between the sources of content, but folders do not have corresponding `created_by` and `created_on` fields to keep track of the same information.

Simply filtering by files may lead to empty folders, there is no clear way to group folders of content by users. Multi-user systems with split data may benefit from this feature, therefore I propose adding these fields.

### Motivation

Systems where multiple users have file upload rights need a way to connect different folders to the identities of the users who created them, independent of the files that may reside in these folders.

This feature was already proposed 2.5 years ago, and it seemed like a good idea at the time, even getting confirmation that PRs were welcome. Since then, formal Feature Requests have become mandatory for PRs implementing new features, and the old one hasn't been converted into the new format.

Original discussion: [Filter folders by created user](https://github.com/directus/directus/discussions/3018)

### Detailed Design

Two fields should be added to Directus Folders:
1. `created_by`, saves the current user ID automatically upon creation
2. `created_on`, saves the current time automatically upon creation

To achieve this, a migration can be used with some reasonable defaults.

### Must Have:
- A field that saves the current user ID whenever a folder is created
- A field that saves the current date and time whenever a folder is created

### Should Have:
- Defaults for the associated interfaces

### Could Have:
- Corresponding fields for tracking updates to the Folder entry (`updated_by`, `updated_on`)

### Won't Have:
- Logic that doesn't already exist in the codebase in some form

### Drawbacks

Implementing this feature as a migration would incur additional overhead on setup, but ensure compatibility.
Additional data would have to be saved to the database, which brings no benefit to single user systems, or systems where users don't typically manage files and/or folders as units of data.
In terms of implementation cost, this feature has a small impact. Adding two fields with logic that already exists elsewhere should not be a particularly difficult task, nor does it require further explanation to the people learning Directus.
This is not a breaking change.

### Alternatives

- Adding the fields by default: would be a more or less forced version of this
- Not adding the fields at all: while it is true that users could implement this for themselves by either adding the fields manually or saving the resulting data elsewhere, consistency demands that folders also keep track of their creators

### Adoption Strategy

A migration suits this usecase, adoption is relatively easy. This feature does not alter any related user workflows, which means that projects that didn't already need this would not need to take additional steps to ensure compatibility.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
